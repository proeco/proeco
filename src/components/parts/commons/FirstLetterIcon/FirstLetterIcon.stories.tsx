import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Box } from '@mui/system';

import { FirstLetterIcon } from './FirstLetterIcon';

export default {
  title: 'parts/commons/FirstLetterIcon',
  component: FirstLetterIcon,
} as ComponentMeta<typeof FirstLetterIcon>;

const Template: ComponentStory<typeof FirstLetterIcon> = ({ ...rest }) => {
  return (
    <Box p="40px" display="flex" alignItems="center" gap="40px">
      <FirstLetterIcon size={40} {...rest} />
      <FirstLetterIcon size={60} {...rest} />
      <FirstLetterIcon size={80} {...rest} />
    </Box>
  );
};

export const Default = Template.bind({});
Default.args = {
  name: 'Proeco',
};

export const LinkFirstLetterIcon = Template.bind({});
LinkFirstLetterIcon.args = {
  name: 'Proeco',
  isLink: true,
  linkUrl: '/',
};
