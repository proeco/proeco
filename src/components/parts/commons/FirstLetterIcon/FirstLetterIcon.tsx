import React, { VFC, ComponentProps } from 'react';
import { Avatar } from '@mui/material';
import { styled } from '@mui/system';

type FirstLetterIconType = {
  name: string;
  size?: number;
};

type Props = ComponentProps<typeof Avatar> & FirstLetterIconType;

export const FirstLetterIcon: VFC<Props> = ({ name, size = 40 }) => {
  return <StyledAvatar size={size}>{name[0]}</StyledAvatar>;
};

const StyledAvatar = styled(Avatar)<{ size: number }>`
  width: ${(props) => props.size}px;
  height: ${(props) => props.size}px;
  font-size: ${(props) => (props.size / 40) * 1.25}rem;
`;
