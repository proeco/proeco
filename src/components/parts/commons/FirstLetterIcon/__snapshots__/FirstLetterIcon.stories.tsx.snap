// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots parts/commons/FirstLetterIcon Default 1`] = `
<div
  className="MuiBox-root css-1le13td"
>
  <div
    className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-qo02ar-MuiAvatar-root"
    size={40}
  >
    P
  </div>
  <div
    className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1x83kw5-MuiAvatar-root"
    size={60}
  >
    P
  </div>
  <div
    className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1y3msk7-MuiAvatar-root"
    size={80}
  >
    P
  </div>
</div>
`;

exports[`Storyshots parts/commons/FirstLetterIcon Link First Letter Icon 1`] = `
<div
  className="MuiBox-root css-1le13td"
>
  <a
    className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-e3mlp-MuiTypography-root-MuiLink-root"
    href="/"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    rel="noopener noreferrer"
    target="_self"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-qo02ar-MuiAvatar-root"
      size={40}
    >
      P
    </div>
  </a>
  <a
    className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-e3mlp-MuiTypography-root-MuiLink-root"
    href="/"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    rel="noopener noreferrer"
    target="_self"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1x83kw5-MuiAvatar-root"
      size={60}
    >
      P
    </div>
  </a>
  <a
    className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-e3mlp-MuiTypography-root-MuiLink-root"
    href="/"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    rel="noopener noreferrer"
    target="_self"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1y3msk7-MuiAvatar-root"
      size={80}
    >
      P
    </div>
  </a>
</div>
`;
